### PR TITLE
RoleAuthority, GameTicker code refactor. Squad assignment improvements.

### PR DESCRIPTION
**Pull request by DominikPanic**
:cl: LaKiller
add: You can now choose a strict "Prefer squad over role" preference that will do anything to put you in your preferred squad first, potentially ignoring your rolled role.
refactor: RoleAuthority, GameTicker code refactor.
/:cl:

### CODE REVIEW

**Comment by MACIEKBAKI** [comment]
despacito?

**Comment by spookydonut** [comment]
The alt_title removal should be a separate pr, start with that.

**Comment by Surrealaser** [comment]
I never knew we needed this but sounds cool.

File: code/game/jobs/role_authority.dm [hunk 1]
```diff
@@ -170,186 +176,142 @@ var/global/datum/authority/branch/role/RoleAuthority
...
-
-*/
-
-	//===============================================================\\
-	//PART III: Here we're doing the main body of the loop and assigning everyone.
-
-	var/l = 0 //levels
-
-	while(++l < 4) //Three macro loops, from high to low.
+	while(++l <= 3) //Three macro loops, from high to low.
```
Comment (Rohesie) [inline code/game/jobs/role_authority.dm]: A `for(var/l = 1 to 3)` works as well.

**Pull request merged**